### DISCUSSION TITLE
Access a relationship to make a filter with it

### DISCUSSION THREAD

**Discussion by BenOussama180** [Power-Components/livewire-powergrid#855] [Q&A]
so i have Links table and orders table with link has many orders by foreign key link_id in orders table.
 i have this in my Link Model  : 
```
protected $appends = [
        'is_on_order',
    ];
public function getIsOnOrderAttribute()
    {
        return $this->orders->where('customer_id', auth()->id())->isEmpty();
    }
```
Datasurce : 

```
public function datasource(): ?Builder
    {
        return Link::whereShowToCustomers(1)
            ->with('orders', 'orders.projects')
            ->groupBy('site');
    }
```

but since i dont perform a ->get() in my datasource i cant access the $link->is_on_order since i want to make a Boolean  filter with it to see if the link has been ordered by the current customer or no, ( doing ->get() makes table so sloow its not an option ) , how can i solve this ? @luanfreitasdev

**Comment by luanfreitasdev** [comment]
How do you think about making the boolean filter? using makeBooleanFilter ?

**Reply by BenOussama180** [reply]
yes exactly : 
```
Column::add()
                ->title('Purchased Before')
                ->field('purchased_before_field', 'is_on_order')
                ->bodyAttribute('text-center text-sm font-[400]', '')
                ->headerAttribute('text-center', ''),
            // ->makeBooleanFilter('is_on_order', 'No', 'Yes'),
```

it works with a " ->get() " since it appends it to the model, but with querybuilder it does not work it says that there is no sucjh 'is_on_order' field on the links table
